core/scripts/cre/environment/environment/workflow.go

```diff
@@ -34,17 +33,13 @@ const (
 )
 
 // getWorkflowRegistryTypeVersion returns the appropriate TypeAndVersion based on the contracts version flag
-func getWorkflowRegistryTypeVersion(contractsVersion string) deployment.TypeAndVersion {
+func getWorkflowRegistryTypeVersion(contractsVersion string) *semver.Version {
 	switch strings.ToLower(contractsVersion) {
 	case "v2":
-		return deployment.TypeAndVersion{
-			Version: *semver.MustParse(creconfig.WorkflowRegistryV2Semver),
-		}
+		return creconfig.WorkflowRegistryV2Semver
```

Comment:
I do think we could leverage the datastore `AddressRef` to use the type and version abstraction, as each `AddressRef` has a `Type` and a `Version` value.

Reply:
So maybe we can build it from there?